helix-term/src/commands.rs

```diff
@@ -1133,7 +1136,14 @@ fn split_selection_on_newline(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
-fn search_impl(doc: &mut Document, view: &mut View, contents: &str, regex: &Regex, extend: bool) {
+fn search_impl(
+    doc: &mut Document,
+    view: &mut View,
+    contents: &str,
+    regex: &Regex,
+    extend: bool,
+    reverse: bool,
```

Comment:
Rather than `reverse: bool`, please use the `Direction` enum that's already present. 

https://github.com/helix-editor/helix/blob/5e6e151d3550f666cd1406f9287398e957f5541b/helix-term/src/commands.rs#L405

We can also replace extend with `Movement::Move/Extend` enum